docs/notebooks/distributed/custom_loop_pjit_example.md

```diff
@@ -0,0 +1,375 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 2
+  status: ok
+  timestamp: 1671545299393
+  user:
+    displayName: Felipe Llinares
+    userId: 01655756739108476525
+  user_tz: -60
+id: DwcrPoWd4r4I
+---
+# Copyright 2022 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+```
+
++++ {"id": "gmlhojcwL5oT"}
+
+`jax.experimental.pjit` example using JAXopt.
+=========================================
+The purpose of this example is to illustrate how JAXopt solvers can be easily
+used for distributed training thanks to `jax.experimental.pjit`. In this case, we begin by
+implementing data parallel training of a multi-class logistic regression model
+on synthetic data.
+
+**NOTE: `jax.experimental.pjit` is not yet supported on Google Colab. Please connect to Google Cloud TPUs to execute the example.**
+
+```{code-cell}
+---
+cellView: form
+executionInfo:
+  elapsed: 76
+  status: ok
+  timestamp: 1671545299595
+  user:
+    displayName: Felipe Llinares
+    userId: 01655756739108476525
+  user_tz: -60
+id: 3Ck_x_LyL7PE
+---
+#@markdown The number of optimization steps to perform:
+MAXITER = 100  #@param{type:"integer"}
+#@markdown The number of samples in the (synthetic) dataset:
+NUM_SAMPLES = 50000  #@param{type:"integer"}
+#@markdown The number of features in the (synthetic) dataset:
+NUM_FEATURES = 784  #@param{type:"integer"}
+#@markdown The number of classes in the (synthetic) dataset:
+NUM_CLASSES = 10  #@param{type:"integer"}
+#@markdown The stepsize for the optimizer (set to 0.0 to use line search):
+STEPSIZE = 0.0  #@param{type:"number"}
+#@markdown The line search approach (either `'zoom'` or `backtracking`), ignored if `STEPSIZE > 0.0`:
+LINESEARCH = 'zoom'  #@param{type:"string"}
+```
+
++++ {"id": "bZmsaMOcONjQ"}
+
+# Imports and TPU setup
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 41809
+  status: ok
+  timestamp: 1671545341521
+  user:
+    displayName: Felipe Llinares
+    userId: 01655756739108476525
+  user_tz: -60
+id: JJuP-Wz_MBeJ
+---
+import functools
+import time
+from typing import Any, Callable, Tuple, Union
+
+from absl import app
+from absl import flags
+
+from flax import jax_utils
+from flax.training import common_utils
```

Comment:
both jax_utils and common_utils are unused

Reply:
Idem.